main/mbcs.c

```diff
@@ -0,0 +1,117 @@
+/*
+*   $Id$
+*
+*   Copyright (c) 2015, vim-jp
+*
+*   This source code is released for free distribution under the terms of the
+*   GNU General Public License.
+*
+*   This module contains functions for checking multibyte character set.
+*/
+
+/*
+*   INCLUDE FILES
+*/
+#define __USE_GNU
+#include "general.h"  /* must always come first */
+
+#ifdef HAVE_ICONV
+
+#include <stdio.h>
+#include <string.h>
+#include <iconv.h>
+#include <errno.h>
+#include "options.h"
+#include "mbcs.h"
+#include "routines.h"
+
+static iconv_t iconv_fd = (iconv_t) -1;
+
+extern boolean openConverter (char* inputEncoding, char* outputEncoding)
+{
+	if (!inputEncoding || !outputEncoding)
+	{
+		static boolean warn;
+		if (!warn)
+		{
+			if (outputEncoding)
+			{
+				error (WARNING, "--input-encoding is not specified");
+				warn = TRUE;
+			}
+			if (inputEncoding)
```

Comment:
I get what it's here for, but the message says "input encoding not specified" yet the test is "output encoding".  This only works because you know that at this point either input our output encoding is NULL, but not both -- so IMO, it'd be more clear and simple to just perform the actual test the warning is about.